File: lab2/AyE2.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
float a,x,g,f,y;
int l, d;
float b=0;

printf("Vvedite x: ");
scanf ("%f", &x);
printf("Vvedite a ");
scanf ("%f", &a);
printf("1-vich G, 2-vich f, 3-vich y");
scanf("%d", &d);
switch(d)
{
case 1:
 if (x>0 && a>0)
 {
 g=9*(-2*a*a+3*a*x+2*x*x);
 l=(6*a*a+29*a*x+35*x*x);
 if (g/l ==0)
 {
 printf( "oshibka");
 exit(1);
 }
 else
 {
 g=g/l;
 printf("G = %f\n,", g);
 exit(1);
 }
 }
 else
 {
 printf( "oshibka znacheniy");
 exit(1);
 }
case 2:
 f=tan(10*a*a-13*a*x+4*x*x);
 printf("F =%f\n", f);
 exit(1);
case 3:
 y=atan(2*a*a+25*a*x+63*x*x);
 printf("Y =%f\n", y);
 exit(1);
 }
/*default:
printf("oshibka")
exit(1);*/
}
/*
printf("Vvedite x ");
scanf ("%f", &x);
printf("Vvedite a ");
scanf ("%f", &a);
f=tan(10*a*a-13*a*x+4*x*x);
printf("f=%f\n\n", f);


printf("Vvedite x ");
scanf ("%f", &x);
printf("Vvedite a ");
scanf ("%f", &a);
y=atan(2*a*a+25*a*x+63*x*x);
printf("y=%f\n\n", y);
}
}
*/

File: lab2/AyE2.py
from numpy import *
from math import *
x = int(input("Введите x: "))
a = int(input("Введите а: " ))
p = int(input("1 - вычислить функцию G, 2 - вычислить функцию F, 3 - вычислить функцию Y: "))
b = 0
if p == 1:
    if (x>0) and (a>0):
        g=9*(-2*a*a+3*a*x+2*x*x)
        l=(6*a*a+29*a*x+35*x*x)
    if math.isclose(l, b, abs_tol = 0.0000001):
        print("Ошибка")
    else:
        g = q/l
    print("Переменная G = " + str(g))
elif p == 2:
    f=tan(10*a*a-13*a*x+4*x*x);
    print("Переменная F = " + str(f))
elif p == 3:
    y=atan(2*a*a+25*a*x+63*x*x);
    print("Переменная Y = " + str(y))
else:
    print("Неверный номер функции")

